features/support/commits_finder.rb

```diff
@@ -0,0 +1,102 @@
+# Makes finding Git commits easier
+class CommitsFinder
+
+  # commit_attributes - the attributes we want to gather for the commits found
+  #                     The default value is the minimalistic set
+  #                     for checking if commits exist at all
+  #
+  # Examples
+  #
+  #   CommitsFinder.new [:branch, :location, :message, :file_name, :file_content, :author]
+  def initialize commit_attributes = [:message]
+    @commit_attributes = commit_attributes.map(&:upcase)
+
+    # The currently known commits
+    @commits = {}
+  end
+
+
+  # Adds the given commit to this CommitsFinder instance
+  #
+  # rubocop:disable MethodLength
+  # rubocop:disable AbcSize
+  def add_commit sha:, message:, branch_name:, author:
+    local_branch_name = local_branch_name branch_name
+    @commits[local_branch_name] ||= {}
+    if @commits[local_branch_name].key? sha
+      # We already have this commit in a different location --> just append the location to the existing commit
+      @commits[local_branch_name][sha]['LOCATION'] << branch_location(branch_name)
+      return
+    end
+
+    commit_data = {
+      'BRANCH' => local_branch_name,
+      'LOCATION' => [branch_location(branch_name)],
+      'MESSAGE' => message
+    }
+    if @commit_attributes.include? 'FILE NAME'
```

Comment:
How about a `has_attribute?` so we can use
```
if has_attribute? :file_name
```
And the method checks if the attributes includes a `file name` whatever case and whether or not its a string on symbol